docs/reference/database-indexes.md

```diff
@@ -0,0 +1,390 @@
+# Database Indexes for Encrypted Columns
+
+EQL supports PostgreSQL B-tree indexes on `eql_v2_encrypted` columns to improve query performance. This guide explains how to create and use indexes effectively.
+
+## Table of Contents
+
+- [Creating Indexes](#creating-indexes)
+- [Index Usage Requirements](#index-usage-requirements)
+- [Query Patterns That Use Indexes](#query-patterns-that-use-indexes)
+- [Query Patterns That Don't Use Indexes](#query-patterns-that-dont-use-indexes)
+- [Index Limitations](#index-limitations)
+- [Best Practices](#best-practices)
+
+---
+
+## Creating Indexes
+
+### Basic Index Creation
+
+Create a B-tree index on an encrypted column using the `eql_v2.encrypted_operator_class`:
+
+```sql
+CREATE INDEX ON table_name (encrypted_column eql_v2.encrypted_operator_class);
+```
+
+**Named index:**
+
+```sql
+CREATE INDEX idx_users_email ON users (encrypted_email eql_v2.encrypted_operator_class);
+```
+
+### When to Create Indexes
+
+Create indexes on encrypted columns when:
+- The table has a significant number of rows (typically > 1000)
+- You frequently query by equality on that column
+- Query performance is important
+- The column contains searchable index terms (hmac_256, blake3, or ore)
+
+---
+
+## Index Usage Requirements
+
+For PostgreSQL to use an index on encrypted columns, **all** of these conditions must be met:
+
+### 1. Column Must Have Appropriate Search Terms
+
+The encrypted data must contain the index term types that support the operation:
+
+- **Equality queries** - Require `unique` index config (adds `hm` hmac_256 or `b3` blake3 terms)
+- **Range queries** - Require `ore` index config (adds `ob` ore_block_u64_8_256 terms)
+- **Pattern matching** - Typically scans (bloom filters don't use B-tree indexes)
+
+**Example:**
+```sql
+-- This data HAS hmac_256 term - index will be used
+'{"i":{"t":"users","c":"email"},"v":2,"hm":"abc123..."}'
+
+-- This data has ONLY bloom filter - index WON'T be used for equality
+'{"i":{"t":"users","c":"email"},"v":2,"bf":[1,2,3]}'
+```
+
+### 2. Index Must Be Created AFTER Data Contains Required Terms
+
+If you:
+1. Insert data without a search term (e.g., only `bf`)
+2. Add the search term later (e.g., add `hm`)
+3. Create an index
+
+**The index will NOT work** until you:
+- Recreate the index, OR
+- Truncate and repopulate the table
+
+**Correct order:**
```

Comment:
What happens when we are using Protect? Do those functions do something special?